Framework/Core/src/DataSampling.cxx

```diff
@@ -134,13 +128,8 @@ std::vector<InputSpec> DataSampling::InputSpecsForPolicy(ConfigurationInterface*
         LOG(WARNING) << "InputSpecsForPolicy does not support subscriptions to all subSpecs yet.";
       }
       for (const auto& path : policy.getPathMap()) {
-        inputs.push_back(
-          InputSpec{
-            path.second.binding.value,
-            path.second.origin,
-            path.second.description,
-            path.second.subSpec,
-            path.second.lifetime });
+        InputSpec input = DataSpecUtils::matchingInput(path.second);
```

Comment:
To fix the failing Data Sampling test, you probably need to overwrite the input.binding with `path.second.binding`, since you add "-matcher" inside `matchingInput` function.